### DISCUSSION TITLE
Do you have to manually unsubscribe from subscriptions?

### DISCUSSION THREAD

**Discussion by SteveMelons** [urql-graphql/urql#2200] [Q&A] [answered]
This might be a silly question but I could not find concrete information on this anywhere.

When you open a subscription on component mount, do you have to ever manually unsubscribe using pause, for example on dismount?

I am wondering this because I have been having some trouble keeping subscriptions alive in React Native when the app state changes from background to foreground or when it just has been running for a while. I thought it might have something to do with subscriptions being duplicated.

**Comment by kitten** [comment, marked as answer]
No, like with queries, as soon as your component unmounts the subscription will be terminated/cancelled.
It's important to note that the pattern described in the "Architecture" section of the docs applies to all operations: https://formidable.com/open-source/urql/docs/architecture/#requests-and-operations-on-the-client
(With mutations being the only "special" operation)

So, as long as you use the bindings, this will all be taken care of for you.

However, Websockets connections themselves may be a different topic. Depending on how your WebSockets/Subscriptions client is implemented, `urql` only "forwards" operations to it, as described on the Subscriptions page, so managing the WebSockets/Subscriptions client is up to you, basically

**Comment by DoisKoh** [comment]
> This might be a silly question but I could not find concrete information on this anywhere.
> 
> When you open a subscription on component mount, do you have to ever manually unsubscribe using pause, for example on dismount?
> 
> I am wondering this because I have been having some trouble keeping subscriptions alive in React Native when the app state changes from background to foreground or when it just has been running for a while. I thought it might have something to do with subscriptions being duplicated.

Take note that I'm using SvelteKit with `@urql/core` and I've had issues with subscriptions getting repeated/stacked when I don't call `.unsubscribe()` - I believe my issue was when I did client-side navigation with SvelteKit, my pages that shared components with the previous page were not properly being unmounted, but were having their `onMount` functions called again. The problem was fixed by calling unsubscribe in the components' `onDestroy` life cycle hooks.
